lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -283,7 +283,7 @@ def content_type
       end
 
       def success_from(response)
-        !response['errorId'] && response['status'] != 'REJECTED'
+        !response['errorId'] && !['REJECTED', 'CANCELLED'].include?(response['status']) && response.dig('status_category') != 'UNSUCCESSFUL'
```

Comment:
Just realized I don't actually have a test that excercises getting to the final check, I'll add one tomorrow.

Reply:
Would it be better to invert this and check all known success states? https://epayments-api.developer-ingenico.com/s2sapi/v1/en_US/java/statuses.html?paymentPlatform=ALL

Reply:
Yeah I think we'll have to do something like that. Their status system is terribly unintuitive. I'm going to close this out for another approach.